File: favorite_places/FAV-2/lib/data/place_store.dart
import 'package:flutter/foundation.dart';
import '../models/place.dart';

class PlaceStore extends ChangeNotifier {
  final List<Place> _places = [
    Place(id: 1, name: 'Mount Fuji', description: 'Iconic volcano in Japan'),
    Place(id: 2, name: 'Eiffel Tower', description: 'Famous landmark in Paris'),
    Place(id: 3, name: 'Grand Canyon', description: 'Impressive natural formation in the USA'),
    Place(id:4, name: "LostCity", description: "This city doesn't exist in any other cities record")
  ];

  List<Place> get places => List.unmodifiable(_places);

  void addPlace(String name, String description) {
    final newPlace = Place(
      id: (_places.isNotEmpty ? _places.map((p) => p.id).reduce((a, b) => a > b ? a : b) : 0) + 1,
      name: name,
      description: description,
    );
    _places.add(newPlace);
    notifyListeners();
  }
}

File: favorite_places/FAV-1/composeApp/src/commonMain/kotlin/website/impulsadigital/data/PlaceStore.kt
package website.impulsadigital.data

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import website.impulsadigital.model.Place
object PlaceStore {
    private val _places = MutableStateFlow(
        listOf(
            Place(1, "Mount Fuji", "Iconic volcano in Japan"),
            Place(2, "Eiffel Tower", "Famous landmark in Paris"),
            Place(3, "Grand Canyon", "Impressive natural formation in the USA")
        )
    )

    val places: StateFlow<List<Place>> = _places

    fun addPlace(name: String, description: String) {
        val current = _places.value
        val newPlace = Place(
            id = (current.maxOfOrNull { it.id } ?: 0) + 1,
            name = name,
            description = description
        )
        _places.value = current + newPlace
    }

    fun getPlacesSnapshot(): List<Place> = places.value
}